clippy_lints/src/copies.rs

```diff
@@ -365,6 +371,38 @@ fn eq_stmts(
         .all(|b| get_stmt(b).map_or(false, |s| eq.eq_stmt(s, stmt)))
 }
 
+fn contains_acceptable_macro(cx: &LateContext<'_>, block: &Block<'_>) -> bool {
+    for stmt in block.stmts {
+        match stmt.kind {
+            StmtKind::Semi(semi_expr) if acceptable_macro(cx, semi_expr) => return true,
+            _ => {},
+        }
+    }
+
+    if let Some(block_expr) = block.expr
+        && acceptable_macro(cx, block_expr)
+    {
+        return true
+    }
+
+    false
+}
+
+fn acceptable_macro(cx: &LateContext<'_>, expr: &Expr<'_>) -> bool {
+    if let ExprKind::Call(call_expr, _)  = expr.kind
```

Comment:
It would be better to not depend on the implementation of the macro. Just checking the macro backtrace would be sufficient.